python/ray/tune/execution/checkpoint_manager.py

```diff
@@ -98,12 +99,10 @@ def newest_persistent_checkpoint(self):
         )
 
     @property
-    def newest_checkpoint(self):
-        """Returns the newest checkpoint (based on training iteration)."""
-        newest_checkpoint = max(
-            [self.newest_persistent_checkpoint, self.newest_memory_checkpoint],
-            key=lambda c: c.id,
-        )
+    def checkpoint(self):
+        """Returns the newest checkpoint."""
+        checkpoints = [self.newest_memory_checkpoint, self.newest_persistent_checkpoint]
```

Comment:
This implicitly gives preference to the memory checkpoint with the same ID over a persistent checkpoint, and I believe we would like to keep the original behavior here 
```suggestion
        checkpoints = [self.newest_persistent_checkpoint, self.newest_memory_checkpoint]
```

Reply:
Yes, I think this is fine to revert. I remember intentionally making the switch to prefer the memory checkpoint because the exploited checkpoint in memory sometimes had the exact same ID as the persistent one. However, the exploited checkpoint should not have the same ID now due to being set as the newest by the checkpoint manager.